terraform/telemetry.py

```diff
@@ -0,0 +1,39 @@
+from google.cloud import pubsub_v1
+from datetime import datetime
+import sys
+
+project_id = "stackdriver-sandbox-230822"
+topic_id = "telemetry"
+
+publisher = pubsub_v1.PublisherClient()
+# The `topic_path` method creates a fully qualified identifier
+# in the form `projects/{project_id}/topics/{topic_id}`
+topic_path = publisher.topic_path(project_id, topic_id)
+
+# define the following helper functions
+def get_datetime_str():
+    # send current date and time for end of installation script
+    now = datetime.now()
+    data = now.strftime("%m/%d/%Y %H:%M:%S")
+    return data
+
+def send_message(args, debug=False):
+    # send in json format
+    data = "{\n\"date\":\"" + get_datetime_str() + "\","
+    
+    # First arg in args is name of script, 'telemetry.py', so we skip it
+    for i in range(len(args) - 1):
+        key_value = args[i + 1].split("=")
+        data += "\n\"" + key_value[0] + "\":\"" + key_value[1]  + "\","
```

Comment:
This won't work for privacy reasons. You need to obfuscate project_id and in this script you don't treat PII differently from other parameters.

Reply:
Another issue: there is no pre-set schema. This means that users can invoke this script manually, sending a bunch of custom key-value pairs that might be read by the server side of it and produce a lot of fake data into our telemetry pipeline. We need to tighten this script to only accept payloads of a particular schemas that we support.

Reply:
Okay. I'll look into obfuscating project id. I'll implement current schema to be:
- date-time
- UUID
- obfuscated project id
- event

Reply:
You might also want to send obfuscated userid, so we can track the # of unique users trying to use it and accomplish different tasks.